rust/ql/lib/codeql/rust/elements/Call.qll

```diff
@@ -2,8 +2,28 @@
  * This module provides the public class `Call`.
  */
 
+private import rust
 private import internal.CallImpl
-
-final class ArgumentPosition = Impl::ArgumentPosition;
+private import internal.CallExprImpl::Impl as CallExprImpl
 
 final class Call = Impl::Call;
+
+final class MethodCall = Impl::MethodCall;
+
+/**
+ * A call expression that targets a closure.
+ *
+ * Closure calls never have a static target, and the set of potential
+ * run-time targets is only available internally to the data flow library.
+ */
+class ClosureCallExpr extends CallExprImpl::CallExprCall {
```

Comment:
The use of closure here is not fully accurate. These calls could also target function pointers, which are not closures.

I think we should

* Change the name to something else, maybe `DynamicCallExpr`.
* Move this class into a file in `‎rust/ql/lib/codeql/rust/dataflow/internal` and name it `LambdaCallExpr` where "lambda" is understood to be the data flow library term and not a Rust term.